dev/benchmarks/macrobenchmarks/lib/src/web/bench_complex_layout.dart

```diff
@@ -0,0 +1,716 @@
+        ),
+        Text(title),
+      ],
+    );
+  }
+}
+
+class MiniIconWithText extends StatelessWidget {
+  const MiniIconWithText(this.icon, this.title);
+
+  final IconData icon;
+  final String title;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      children: <Widget>[
+        Padding(
+          padding: const EdgeInsets.only(right: 8.0),
+          child: Container(
+            width: 16.0,
+            height: 16.0,
+            decoration: BoxDecoration(
+              color: Theme.of(context).primaryColor,
+              shape: BoxShape.circle,
+            ),
+            child: Icon(icon, color: Colors.white, size: 12.0),
+          ),
+        ),
+        Text(title, style: Theme.of(context).textTheme.caption),
+      ],
+    );
+  }
+}
+
+class FatDivider extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: 8.0,
+      color: Theme.of(context).dividerColor,
+    );
+  }
+}
+
+class UserHeader extends StatelessWidget {
+  const UserHeader(this.userName);
+
+  final String userName;
+
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          const Padding(
+            padding: EdgeInsets.only(right: 8.0),
+            child: Text('placeholder')
+          ),
+          Expanded(
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.start,
+              crossAxisAlignment: CrossAxisAlignment.stretch,
+              children: <Widget>[
+                RichText(text: TextSpan(
+                  style: Theme.of(context).textTheme.bodyText2,
+                  children: <TextSpan>[
+                    TextSpan(text: userName, style: const TextStyle(fontWeight: FontWeight.bold)),
+                    const TextSpan(text: ' shared a new '),
+                    const TextSpan(text: 'photo', style: TextStyle(fontWeight: FontWeight.bold)),
+                  ],
+                )),
+                Row(
+                  children: <Widget>[
+                    Text('Yesterday at 11:55 • ', style: Theme.of(context).textTheme.caption),
+                    Icon(Icons.people, size: 16.0, color: Theme.of(context).textTheme.caption.color),
+                  ],
+                ),
+              ],
+            ),
+          ),
+          TopBarMenu(),
+        ],
+      ),
+    );
+  }
+}
+
+class ItemDescription extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return const Padding(
+      padding: EdgeInsets.all(8.0),
+      child: Text('Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'),
+    );
+  }
+}
+
+class ItemImageBox extends StatelessWidget {
```

Comment:
We don't use actual images, do we? (at least, I can't find any in the code) If we do, I'd be curious if the images are loaded synchronously enough for the benchmark to pick them up. Otherwise we may end up in a situation when the benchmark is done and images are still loading.